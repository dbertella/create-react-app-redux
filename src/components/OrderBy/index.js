import React, { PropTypes } from 'react';

const OrderBy = ({sortArtist, sortParam}) => (
  <div>
    OrderBy:
    <button onClick={(e) => sortArtist('age', sortParam.age)}>Age {sortParam.age}</button>
    <button onClick={(e) => sortArtist('rate', sortParam.rate)}>Rate {sortParam.rate}</button>
  </div>
);

OrderBy.propTypes = {
  sortArtist: PropTypes.func.isRequired,
  sortParam: PropTypes.object.isRequired,
}

export default OrderBy;
