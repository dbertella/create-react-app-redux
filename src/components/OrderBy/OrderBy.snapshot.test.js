import React from 'react';
import OrderBy from './';
import renderer from 'react-test-renderer';

const sortParam = {
  age: 'ASC',
  rate: 'DESC',
};
const stubFunc = () => 'app';

it('initial render', () => {
  const component = renderer.create(
    <OrderBy sortArtist={stubFunc} sortParam={sortParam} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
