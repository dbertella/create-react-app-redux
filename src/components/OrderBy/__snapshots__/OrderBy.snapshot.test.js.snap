exports[`test initial render 1`] = `
<div>
  OrderBy:
  <button
    onClick={[Function onClick]}>
    Age 
    ASC
  </button>
  <button
    onClick={[Function onClick]}>
    Rate 
    DESC
  </button>
</div>
`;
